sst/packages/functions/src/cron-jobs/api-key-expiry/main.ts

```diff
@@ -0,0 +1,93 @@
+import { eq, sql } from 'drizzle-orm'
+import { Config } from 'sst/node/config'
+
+import PostmanDbClient from '@/core/database/client'
+import { apiKeys } from '@/core/models'
+import { users } from '@/core/models/user'
+import { getFutureUTCDate } from '@/core/util/date'
+import { sendEmail } from '@/core/util/email'
+
+import { IS_LOCAL, LOCAL_DB_URI } from '../../env'
+
+import { reminderEmailMapper } from './helper'
+
+/*
+- This function is triggered by a cron job that runs every day at 12AM UTC, i.e. 8AM SGT
+- It picks up API keys that are expiring 28 days, 14 days, 3 days and 1 day from now
+- DATE_TRUNC("day", valid_until) function round down the valid_until timestamp to the nearest day
+- The cron job will send emails to the API key owners to remind them to renew their API keys
+*/
+
+export async function handler() {
```

Comment:
Can we wrap this function body around a try/catch and log the error out before letting it bubble and fail cronitor?

Reply:
good idea